src/kOS/Communication/MessageStructure.cs

```diff
@@ -41,10 +42,22 @@ private void InitializeSuffixes()
         {
             AddSuffix("SENTAT", new Suffix<kOS.Suffixed.TimeSpan>(() => new kOS.Suffixed.TimeSpan(Message.SentAt)));
             AddSuffix("RECEIVEDAT", new Suffix<kOS.Suffixed.TimeSpan>(() => new kOS.Suffixed.TimeSpan(Message.ReceivedAt)));
-            AddSuffix("SENDER", new Suffix<VesselTarget>(() => Message.Vessel));
+            AddSuffix("SENDER", new Suffix<VesselTarget>(GetVesselTarget));
             AddSuffix("CONTENT", new Suffix<Structure>(DeserializeContent));
         }
 
+        public VesselTarget GetVesselTarget()
+        {
+            Vessel vessel = FlightGlobals.Vessels.Find((v) => v.id.ToString().Equals(Message.Vessel));
+
+            if (vessel == null)
+            {
+                throw new KOSCommunicationException("Vessel does not exist");
+            }
+
+            return new VesselTarget(vessel, shared);
+        }
+
```

Comment:
I think given that this can throw exception that breaks the execution, we need protection against that somehow.  I don't see a way for the script to detect if the vessel it's getting no longer exists.  Using the pattern used elsewhere, that would mean having a `HASSENDER` boolean suffix.